packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -83,8 +83,19 @@ export default class Proxy {
     return id
   }
 
-  unregisterProxy(id) {
-    return this._db.remove(id)
+  async unregisterProxy(id) {
+    const { vmUuid } = await this._getProxy(id)
+
+    await this._db.remove(id)
+
+    if (vmUuid !== undefined) {
+      this._app
```

Comment:
Don't we want to `await` this as well? It could make sense if we want to trigger a refresh of the licenses in the UI, for example.

Reply:
We probably don't want to fail if the proxy has been deleted.

Reply:
Right, but the `.catch()` makes it impossible to fail.

(I should have put my comment on the whole block :roll_eyes:)

Reply:
Makes sense indeed.